groups/infrastructure/data.tf

```diff
@@ -0,0 +1,31 @@
+data "vault_generic_secret" "secrets" {
+  path = "applications/${var.aws_profile}/${var.environment}/${local.stack_fullname}"
+}
+
+data "aws_kms_key" "stack_configs" {
+  key_id = "alias/${var.aws_profile}/${local.kms_key_alias}"
+}
+
+data "aws_subnets" "application" {
+  filter {
+    name   = "tag:Name"
+    values = [local.application_subnet_pattern]
+  }
+  filter {
+    name   = "tag:NetworkType"
+    values = ["private"]
+  }
```

Comment:
This filter shouldn't be required as we're feeding in a known subnet name pattern which should always be the application subnets